afs-core/src/main/java/com/powsybl/afs/AbstractNodeBase.java

```diff
@@ -202,19 +206,21 @@ public void archive(Path dir, boolean useZip) {
 
     public void unarchive(Path dir, boolean isZipped) {
         if (isZipped) {
+            int index = dir.toString().lastIndexOf('.');
+            Path nodeDir = Paths.get(dir.toString().substring(0, index));
             try {
-                int index = dir.toString().lastIndexOf('.');
-                Path nodeDir = Paths.get(dir.toString().substring(0, index));
```

Comment:
It's not necessary to move these lines. Try to make only needed modification when you make a contribution.

Reply:
Ok for next time but can I leave it like this for this time?

Reply:
No